### PR TITLE
Migrating the first core concepts page

### PR DESCRIPTION
**Pull request by spbail**
I started migrating the "core concepts" pages. This is just a first draft to provide a starting point for conversations. Some comments:
- The only fully migrated page is "Checkpoints" - I'd love to clarify a few things both re: formatting and content before moving forward.
- I have added info boxes with "sam-notes" titles (they're rendered in bright blue) with some suggestions for how to change the content
- I also took some general docusaurus notes - might be worth using github (e.g. the wiki or a discussion function) for that.
- I'd suggest a pretty thorough refactor of the core concept section (see my notes) and removing/reorganizing the content - let's chat!

### CODE REVIEW

File: docs/docs/reference/core_concepts/dividing_data_assets_into_batches.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+title: Dividing Assets Into Batches
+---
+
+:::info sam-notes
+I'd suggest renaming this to "batches"
```
Comment (jcampbell) [inline docs/docs/reference/core_concepts/dividing_data_assets_into_batches.md:6]: +1

File: docs/docs/reference/core_concepts/data_discovery.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+title: Data Discovery
+---
+
+:::info sam-notes
+This is basically an explanation of profiling, and pretty short. I suggest moving the content to the profilers page.
```
Comment (jcampbell) [inline docs/docs/reference/core_concepts/data_discovery.md:6]: This grew out of an attempt to differentiate profilers from profiling...i.e. the build suite workflow from the describe data workflow. Would love to figure out a better way to present that distinction.

File: docs/docs/reference/core_concepts/stores.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+title: Stores
+---
+
+:::info sam-notes
+This page doesn't exist, but I think it's a relevant core concept that should have a page?
```
Comment (jcampbell) [inline docs/docs/reference/core_concepts/stores.md:6]: +1

**Review by jcampbell — APPROVED** [review]
A couple comments inline

**Comment by spbail** [comment]
I'm abandoning this in favor of https://github.com/great-expectations/great_expectations/pull/2810

**Pull request closed without merging**